compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -1486,25 +1486,21 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                     let kind = key.descr();
                     let mut returned_async_output_error = false;
                     for sp in values {
-                        err.span_label(
-                            *sp,
-                            format!(
-                                "{}{}{} {}{}",
-                                if sp.is_desugaring(DesugaringKind::Async)
-                                    && !returned_async_output_error
-                                {
-                                    "checked the `Output` of this `async fn`, "
-                                } else if count == 1 {
-                                    "the "
-                                } else {
-                                    ""
-                                },
-                                if count > 1 { "one of the " } else { "" },
-                                target,
-                                kind,
-                                pluralize!(count),
-                            ),
-                        );
+                        if sp.is_desugaring(DesugaringKind::Async) && !returned_async_output_error {
+                            err.span_label(*sp, format!("{}", "async functions return futures"));
+                        } else {
+                            err.span_label(
+                                *sp,
+                                format!(
+                                    "{}{}{} {}{}",
+                                    if count == 1 { "the " } else { "" },
+                                    if count > 1 { "one of the " } else { "" },
+                                    target,
+                                    kind,
+                                    pluralize!(count),
+                                ),
+                            );
+                        }
                         if sp.is_desugaring(DesugaringKind::Async)
```

Comment:
This `if` block can now be absorbed into the previous one.

Reply:
Good point! Done!